controllers/compassmanager_controller.go

```diff
@@ -108,7 +119,7 @@ func (cm *CompassManagerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 	cm.Log.Infof("Compass Runtime Agent for Runtime %s configured.", compassRuntimeID)
 
-	return ctrl.Result{}, cm.markRuntimeRegistered(req.NamespacedName, compassRuntimeID)
+	return ctrl.Result{}, cm.createCompassMappingResource(compassRuntimeID, kymaLabels)
```

Comment:
One note: if we decide to provide statuses on the CR we should create it even in case the Compass registration failed. This is just a reminder, I think we should discuss if we go with the status field, and then create a new PR.